### DISCUSSION TITLE
Elastic Load Balancing

### DISCUSSION THREAD

**Discussion by jakubsuchy** [ubicloud/ubicloud#1354] [Ideas]
It would be great to support ELB/NLBs next. While some new LBs are "popular", open source HAProxy is still the gold standard for performance.

**Comment by furkansahin** [comment]
Hey @jakubsuchy , thanks for the input! I will incorporate this into our next quarter planning.

**Comment by ozgune** [comment]
I wanted to share a comment here to associate a related GitHub issue with this discussion: https://github.com/ubicloud/ubicloud/issues/481

**Comment by tsojcanth** [comment]
I'm not specifically attached to any specific software, but here are the features in order of importance:

- SSL Offloading
- support for let's encrypt challenges
- target groups (a-la AWS ELB) with liveliness check
- support for L7 conditionals and responses (a-la AWS ELB, in an ideal world touring complete in a similar manner to F5 BigIP iRules, but that's a lot).
- a non-cli user interface
- support for customizable 4xx and 5xx pages. For example I'd like to have custom 404 and Bad Gateway page.

Things I do not want:

1. WAF. In my experience it makes debug harder, it's often not transparent.
2. that's it

**Comment by ozgune** [comment]
Hey all, @furkansahin is now looking into our elastic load balancer's v1. I think he'll document the initial set of features that we're planning to support in this thread.

If you're planning to use load balancers with Ubicloud, please feel free to share your requirements in this discussion.

**Comment by ozgune** [comment]
Hi @jakubsuchy , @tsojcanth - We recently enabled Network Load Balancers and talked about our implementation here: https://www.ubicloud.com/blog/ubicloud-load-balancer-simple-and-cost-free

You can also check our documentation here: https://www.ubicloud.com/docs/networking/load-balancer

In summary, we provide common load balancing algorithms to balance requests across VMs and provide high availability through health checks. We encrypt all traffic and our basic plan is free. Our current implementation sits at the L4 layer. We'll probably improve on this soon.

@furkansahin is also about to merge certificate related work.

I'll close this issue since we now have an initial implementation. Please feel free to check out our current implementation. If you'd like to see additional improvements, please feel free to start a new discussion about Elastic Load Balancing improvements. Thank you again for your input on this!
